Cubical/Algebra/CommAlgebra/Instances/Initial.agda

```diff
@@ -0,0 +1,76 @@
+{-# OPTIONS --cubical --safe --no-import-sorts #-}
+module Cubical.Algebra.CommAlgebra.Instances.Initial where
+
+open import Cubical.Foundations.Everything
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Sigma.Properties using (Σ≡Prop)
+
+open import Cubical.Algebra.CommRing
+open import Cubical.Algebra.Ring
+open import Cubical.Algebra.Algebra.Base using (IsAlgebraHom)
+open import Cubical.Algebra.CommAlgebra.Base
+
+private
+  variable
+    ℓ : Level
+
+module CommAlgebraExamples ((R , str) : CommRing ℓ) where
+
+  initial : CommAlgebra (R , str) ℓ
+  initial =
+    let open CommRingStr str
+    in  (R , commalgebrastr _ _ _ _ _ (λ r x → r · x)
+                    (makeIsCommAlgebra (isSetRing (CommRing→Ring (R , str)))
+                       +Assoc +Rid +Rinv +Comm
+                       ·Assoc ·Lid
+                       ·Ldist+ ·-comm
+                        (λ x y z → sym (·Assoc x y z)) ·Ldist+ ·Rdist+ ·Lid
+                         λ x y z → sym (·Assoc x y z)))
+
+
+  module _ (A : CommAlgebra (R , str) ℓ) where
+    open CommAlgebraStr ⦃... ⦄
+    instance
+      _ : CommAlgebraStr (R , str) (fst A)
+      _ = snd A
+      _ : CommAlgebraStr (R , str) R
+      _ = snd initial
+
+    _*_ : R → (fst A) → (fst A)
+    r * a = CommAlgebraStr._⋆_ (snd A) r a
+
+    initialMap : CommAlgebraHom initial A
+    initialMap =
+      makeCommAlgebraHom {M = initial} {N = A}
```

Comment:
Copatterns

Reply:
Can I consider this as resolved, now that makeCommAlgebraHom is defined by a copattern?

Reply:
@Saizan : Isn't it actually the case that even if the function that one uses is defined using copatterns that is not going to help here and we should use copatterns here as well?